src/cljs/main/broadfcui/common/table/utils.cljs

```diff
@@ -0,0 +1,82 @@
+(ns broadfcui.common.table.utils
+  (:require
+    [broadfcui.utils :as utils]
+    ))
+
+
+(defn resolve-id [{:keys [header id]}]
+  (assert (or (not id) (string? id)) "Column id must be a string, if present")
+  (assert (or id (string? header)) "Every column must have a string header or id")
+  (or id header))
+
+
+(defn canonical-name [{:keys [header id]}]
```

Comment:
I'll rename and add some docstrings, but `canonical-name` operates on a "resolved column" (i.e. the full column from the definitions), and `resolve-canonical-name` works from a column ID and the list of column definitions.